src/utils/daos_hdlr.c

```diff
@@ -583,13 +634,29 @@ cont_list_snaps_hdlr(struct cmd_args_s *ap)
 	if (expected_count < snaps_count)
 		fprintf(stderr, "size required to gather all snapshots has raised, list has been truncated\n");
 
-	for (i = 0; i < min(expected_count, snaps_count); i++)
-		D_PRINT(DF_U64" ", buf[i]);
-	D_PRINT("\n");
+	if (snapname == NULL) {
+		for (i = 0; i < min(expected_count, snaps_count); i++)
+			D_PRINT(DF_U64" %s\n", epochs[i], names[i]);
+	} else {
+		for (i = 0; i < min(expected_count, snaps_count); i++)
+			if (strcmp(snapname, names[i]) == 0) {
+				*epoch = epochs[i];
```

Comment:
Is it possible for the epoch pointer to be null? I don't see it checked anywhere in this function.

Reply:
hmm, right I wrongly assumed that both snap name and epoch will be either NULL or not together, but you are right it needs to be checked to prevent issue if function will be reused.